packages/flame/lib/src/cache/images.dart

```diff
@@ -45,6 +45,19 @@ class Images {
     _assets[name] = _ImageAsset.fromImage(image);
   }
 
+  /// If the image with [name] exists in the cache that is returned, otherwise
+  /// the image generated by [imageGenerator] is returned.
+  ///
+  /// If the [imageGenerator] is used, the resulting [Image] is stored with
+  /// [name] in the cache.
+  Future<Image> fetchOrGenerate(
+    String name,
+    Future<Image> Function() imageGenerator,
```

Comment:
Why is it named "Generator" instead of "Builder"?

Reply:
I didn't name it, but it's not a builder (if the builder name is used according to the builder pattern).